Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp

```diff
@@ -4131,3 +4131,118 @@ WebKitFeatureList* webkit_settings_get_development_features(void)
+ *
+ * Returns: %TRUE if the settings were correctly applied or %FALSE on error.
+ *
+ * Since: 2.46
+ */
+gboolean webkit_settings_apply_from_key_file(WebKitSettings* settings, GKeyFile* keyFile, const gchar* groupName, GError** error)
+{
+    g_return_val_if_fail(WEBKIT_IS_SETTINGS(settings), FALSE);
+    g_return_val_if_fail(keyFile, FALSE);
+    g_return_val_if_fail(groupName, FALSE);
+    g_return_val_if_fail(!error || !*error, FALSE);
+
+    GObjectClass* klass = G_OBJECT_GET_CLASS(settings);
+    unsigned totalProperties = 0;
+    GUniquePtr<GParamSpec*> properties(g_object_class_list_properties(klass, &totalProperties));
+
+    if (!g_key_file_has_group(keyFile, groupName)) {
+        g_set_error(error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_INVALID_VALUE, "The key file has no %s group", groupName);
+        return FALSE;
+    }
+
+    HashMap<String, std::variant<bool, unsigned, String>> values;
+    for (unsigned i = 0; i < totalProperties; i++) {
+        GUniqueOutPtr<GError> lookupError;
+        auto propertyName = makeString(properties.get()[i]->name);
+        if (!g_key_file_has_key(keyFile, groupName, propertyName.utf8().data(), &lookupError.outPtr())) {
+            if (lookupError) {
+                g_propagate_error(error, lookupError.release());
+                return FALSE;
+            }
+            continue; // Setting missing in GKeyFile, skip it.
+        }
+
+        switch (G_PARAM_SPEC_VALUE_TYPE(properties.get()[i])) {
+        case G_TYPE_BOOLEAN: {
+            bool value = g_key_file_get_boolean(keyFile, groupName, propertyName.utf8().data(), &lookupError.outPtr());
+            if (!value && lookupError) {
+                g_propagate_error(error, lookupError.release());
+                return FALSE;
+            }
+            values.add(WTFMove(propertyName), value);
+            break;
+        }
+        case G_TYPE_UINT: {
+            guint64 value = g_key_file_get_uint64(keyFile, groupName, propertyName.utf8().data(), &lookupError.outPtr());
+            if (!value && lookupError) {
+                g_propagate_error(error, lookupError.release());
+                return FALSE;
+            }
+            if (value > G_MAXUINT) {
+                g_set_error(error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_INVALID_VALUE, "Value for '%s' exceeds maximum integer size", propertyName.utf8().data());
+                return FALSE;
+            }
+            values.add(WTFMove(propertyName), static_cast<unsigned>(value));
+            break;
+        }
+        case G_TYPE_STRING: {
+            GUniquePtr<char> value(g_key_file_get_string(keyFile, groupName, propertyName.utf8().data(), &lookupError.outPtr()));
+            if (!value) {
+                g_assert(lookupError);
+                g_propagate_error(error, lookupError.release());
+                return FALSE;
+            }
+            values.add(WTFMove(propertyName), makeString(value.get()));
```

Comment:
I suggested to use CString instead of string for the value, but maybe we can even use `GUniquePtr<char>` and we don't copy anything?